src/SurfaceFluxes.jl

```diff
@@ -357,50 +363,76 @@ function obukhov_length(
     sc::AbstractSurfaceConditions{FT},
     uft::UF.AUFT,
     scheme;
-    tol::RS.AbstractTolerance = RS.SolutionTolerance(max(FT(1e-1), sqrt(eps(FT)))),
+    tol::RS.AbstractTolerance = RS.SolutionTolerance(FT(z_in(sc) / 50)),
     maxiter::Int = 10,
+    soltype = RS.CompactSolution(),
 ) where {FT}
 
     function root_l_mo(x_lmo)
-        root = x_lmo - local_lmo(param_set, x_lmo, sc, uft, scheme)
-        return root
+        residual = x_lmo - local_lmo(param_set, x_lmo, sc, uft, scheme)
+        return residual
     end
-    sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), RS.CompactSolution(), tol, maxiter)
-    if sol.converged
-        L_MO = sol.root
-    else
-        thermo_params = SFP.thermodynamics_params(param_set)
+
+    thermo_params = SFP.thermodynamics_params(param_set)
+    grav = SFP.grav(param_set)
+    cp_d = SFP.cp_d(param_set)
+    DSEᵥ_sfc = TD.virtual_dry_static_energy(thermo_params, ts_sfc(sc), grav * z_sfc(sc))
+    DSEᵥ_in = TD.virtual_dry_static_energy(thermo_params, ts_in(sc), grav * z_in(sc))
+    ΔDSEᵥ = DSEᵥ_in - DSEᵥ_sfc
+    tol_neutral = FT(cp_d / 10)
+
+    function solve_lmo(ΔDSEᵥ)
+        if abs(ΔDSEᵥ) <= tol_neutral
+            L_MO = z_in(sc) / tol_neutral
+            sol = nothing
+            is_converged = true
+        else
+            sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), soltype, tol, maxiter)
+            L_MO = sol.root
+            is_converged = sol.converged
+        end
+        return sol, L_MO, is_converged
+    end
```

Comment:
Having multiple `is_converged` is confusing.

Also, the tolerance `tol::RS.AbstractTolerance = RS.SolutionTolerance(FT(z_in(sc) / 50))` is not used here: `abs(ΔDSEᵥ) <= tol_neutral`.

Reply:
The tolerance for the statement that verifies whether we have a neutral b.l. is indeed different to that set for the L_MO solver (we're making the neutrality check depend on `cp_d` since this is related to the DSE value, while we specify the absolute tolerance required for convergence in L_MO based on grid spacing). Removing `is_converged` flag in #57.